connaisseur/validators/notaryv1/notaryv1_validator.py

```diff
@@ -237,19 +246,28 @@ def __search_image_targets_for_tag(trust_data: dict, image: Image):
         base64_digest = trust_data[image_tag]["hashes"]["sha256"]
         return base64.b64decode(base64_digest).hex()
 
-    def __update_with_delegation_trust_data(
+    async def __update_with_delegation_trust_data(
         self, trust_data, delegations, key_store, image
     ):
-        for delegation in delegations:
-            delegation_trust_data = self.notary.get_delegation_trust_data(
-                image, TUFRole(delegation)
-            )
-            # when delegations are added to the repository, but weren't yet used for
-            # signing, the delegation files don't exist yet and are `None`. in this
-            # case validation must be skipped
-            if delegation_trust_data is not None:
-                delegation_trust_data.validate(key_store)
-            trust_data[delegation] = delegation_trust_data
+        delegation_trust_data_list = await asyncio.gather(
+            *[
+                self.notary.get_delegation_trust_data(image, TUFRole(delegation))
+                for delegation in delegations
+            ]
+        )
+
+        # when delegations are added to the repository, but weren't yet used for signing,
+        # the delegation files don't exist yet and are `None`. in this case validation
+        # must be skipped
+        delegation_trust_data = {
```

Comment:
This is subtly different from the previous version. Previous `delegation_trust_data` that were `None` were included in `trust_data` while not being validated. Now they are not included and not validated

Haven't really thought this through and maybe you see this more easily. Does that make a difference?

Reply:
Also the comment is now wrong

Reply:
now that i think about it. currently we are downloading ALL delegation trust data, whether we need them or not. i think we should change and only download those we actually need. then wen can also raise an error here, should an required delegation trust data be `None`

Reply:
Sounds reasonable and would also prevent DoS-like attack of spamming trust data. Can't see anything wrong with the idea 👍

Reply:
#171 will be done in a seperate PR